src/mon/MDSMonitor.cc

```diff
@@ -140,6 +138,11 @@ void MDSMonitor::create_pending()
   pending_fsmap = fsmap;
   pending_fsmap.epoch++;
 
+  if (mon->osdmon()->is_readable()) {
```

Comment:
if you have concern about the ordering in which osdmap is updated and mdsmap is updated, as`PAXOS_MDSMAP < PAXOS_OSDMAP`. probably you can do this in `post_paxos_update()` where all paxos services have been refreshed.

Reply:
that ensures a valid osdmap, but we still shouldn't touch the fsmap.  the change should be applied to pending_fsmap, which means it goes in create_pending or encode_pending

Reply:
ahh, agreed.